qiskit/_result.py

```diff
@@ -156,9 +156,10 @@ def get_data(self, name):
         """
         if self._is_error():
             exception = self.__result['result']
-            if isinstance(exception, RegisterSizeError):
-                raise exception # Re-throw RegisterSizeError
-            raise QISKitError(str(exception))
+            # if isinstance(exception, RegisterSizeError):
```

Comment:
This was a debugging aid I accidentally let slip through since the traceback was being dropped. Thanks for the chaining recommendation you mentioned elsewhere. I'll try to implement that.

Reply:
Yeah, actually, leave it like that and just remove the commented code. As I said, I think it's better if we re-throw original exceptions from the jobs, instead of transforming them to QISKitError :)